isolated-buttons/src/Button.js

```diff
@@ -0,0 +1,113 @@
+import {Observable, Subject} from 'rx'
+import isolate from '@cycle/isolate'
+import {div, button} from 'cycle-snabbdom'
+import combineLatestObj from 'rx-combine-latest-obj'
+
+const ADD = id => ({id, type: 'add'})
+const REMOVE = id => ({id, type: 'remove'})
+
+function intent({DOM, action$}, id) {
+  const add$ = DOM.select('.add').events('click').map(() => ADD(id))
+  const remove$ = DOM.select('.remove').events('click').map(() => REMOVE(id))
+  const removeButton$ = action$.filter(action => action.type === 'remove')
+  return {add$, remove$, removeButton$}
+}
+
+function findButton(children, _id) {
+  const id = parseInt(_id)
+  let pointerId;
+  let index;
+  let top = children.length;
+  let bottom = 0;
+  for (let i = children.length - 1; i >= 0; i--) { // binary search
+    index = bottom + ((top - bottom) >> 1);
+    pointerId = parseInt(children[index].id);
+    if (pointerId === id) {
+      return index;
+    } else if (pointerId < id) {
+      bottom = index;
+    } else if (pointerId > id) {
+      top = index;
+    }
+  }
+  return null;
+}
+
+function model(actions, sources, parentId) {
+  const newButtonMod$ = actions.add$.map(() => children => {
```

Comment:
I know `() => children => {` is more compact, but I've been using `() => function(children) {` because it looks more obvious to the average programmer that it returns a function. Like line 92 is good. Even better is to name that function `newButtonMod`. I'd even replace "mod" with "update", like "update" in the Elm arch. Check also https://github.com/futurice/power-ui/blob/master/src/components/pages/PowerheadPage/model.js#L34